src/util/InteractiveMenu/InteractiveMenu.ts

```diff
@@ -0,0 +1,67 @@
+import {
+    ActionRowBuilder,
+    ButtonBuilder,
+    ComponentType,
+    EmbedBuilder,
+    Message,
+    RepliableInteraction,
+} from "discord.js";
+import { Player } from "../Player/Player.js";
+
+export default abstract class InteractiveMenu {
+    protected message?: Message;
+    protected interaction: RepliableInteraction;
+
+    /**
+     * Creates a new PlayerMenu
+     * @param interaction This is the interaction from the command to be able to reply
+     * @param player This is a player object to be able to access stuff within player.
+     */
+    constructor(interaction: RepliableInteraction) {
+        this.interaction = interaction;
+    }
+
+    /**
+     * This will create embeds that will be displayed on the user side
+     * @returns embeds for generateMessagePayload
+     */
+    protected abstract generateEmbeds(): [EmbedBuilder];
+
+    /**
+     * This will create the message to display to the play
+     * @param hideButtons This is to konw if we are hiding the buttons, only will happen if player stays
+     * @returns The object to display with the reply.
+     */
+    protected abstract generateMessagePayload(): {};
```

Comment:
generateMessagePayload should not be abstract as it calls `generateEmbeds()` and `generateComponents()` each and every time to be returned to the `.render()` function.